test/host/scriptDirFile.js

```diff
@@ -0,0 +1,25 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+const __filename = WScript.GetFileName();
+const __dirname = WScript.GetDirectory();
+const {__filename2, __dirname2} = WScript.LoadScriptFile(`${__dirname}\\scriptDirFile_2.js`, "samethread");
+WScript.LoadScriptFile(`${__dirname}\\scriptDirFile_3.js`, "self");
+const {__filename4, __dirname4} = WScript.LoadScriptFile(`${__dirname}\\nestedDir\\scriptDirFile_4.js`, "samethread");
+WScript.LoadScriptFile(`${__dirname}\\nestedDir\\scriptDirFile_5.js`, "self");
+
+if (
+  __dirname !== __dirname2 ||
+  __dirname !== __dirname3 ||
+  __dirname !== __dirname5 ||
+  __filename !== __filename3 ||

```

Comment:
also good to leave a note about why the filename is incorrect in the 3 and 5 case, due to the "self" limitation